scripts/pylib/twister/twisterlib/testsuite.py

```diff
@@ -407,13 +407,17 @@ def add_testcase(self, name, freeform=False):
     def get_unique(testsuite_root, workdir, name):
 
         canonical_testsuite_root = os.path.realpath(testsuite_root)
-        if Path(canonical_zephyr_base) in Path(canonical_testsuite_root).parents:
-            # This is in ZEPHYR_BASE, so include path in name for uniqueness
-            # FIXME: We should not depend on path of test for unique names.
-            relative_ts_root = os.path.relpath(canonical_testsuite_root,
+        relative_ts_root = os.path.relpath(canonical_testsuite_root,
                                                start=canonical_zephyr_base)
-        else:
-            relative_ts_root = ""
+        # Include path in name for name uniqueness.
+        # Replace all '..' with a single `out_of_tree` for out of tree tests.
+        # FIXME: We should not depend on path of test for unique names.
+        is_outside_zephyr_base  = Path(canonical_zephyr_base) not in Path(canonical_testsuite_root).parents
+        if is_outside_zephyr_base:
+            relative_ts_parts = Path(relative_ts_root).parts
+            if ".." in relative_ts_parts:
+                relative_ts_parts = tuple(filter(("..").__ne__, relative_ts_parts))
```

Comment:
Please consider replace it by more pythonic way with list comprehension:
```python
relative_ts_parts = [part for part in relative_ts_parts if part != ".."]
```
or generator:
```python
relative_ts_parts = tuple(part for part in relative_ts_parts if part != "..")
```